comparison between numeric values of different types panic

The following program `sample.go` triggers a panic:

```go
package main

func main() {
    println(2 < 2.5)
}
```

Expected result:
```console
$ go run ./sample.go
true
```

Got:
```console
$ dyngo ./sample.go
panic: reflect: call of reflect.Value.Int on float64 Value [recovered]
	panic: reflect: call of reflect.Value.Int on float64 Value

goroutine 1 [running]:
github.com/containous/dyngo/interp.runCfg.func1(0xc00016c2d0)
	/Users/marc/go/src/github.com/containous/dyngo/interp/run.go:95 +0x134
panic(0x17c11a0, 0xc000162720)
	/usr/local/Cellar/go/1.11.4/libexec/src/runtime/panic.go:513 +0x1b9
reflect.Value.Int(...)
	/usr/local/Cellar/go/1.11.4/libexec/src/reflect/value.go:961
github.com/containous/dyngo/interp.lower.func2(0xc00016c2d0, 0x18be028)
	/Users/marc/go/src/github.com/containous/dyngo/interp/run.go:905 +0x230
github.com/containous/dyngo/interp.runCfg(0xc0001d8ea0, 0xc00016c2d0)
	/Users/marc/go/src/github.com/containous/dyngo/interp/run.go:100 +0x62
github.com/containous/dyngo/interp.(*Interpreter).run(0xc0001da070, 0xc0001d8480, 0xc00016c230)
	/Users/marc/go/src/github.com/containous/dyngo/interp/run.go:82 +0xa6
github.com/containous/dyngo/interp.(*Interpreter).Eval(0xc0001da070, 0xc000166030, 0x30, 0x188998f, 0x4, 0xc0001da070, 0x0, 0x1785820)
	/Users/marc/go/src/github.com/containous/dyngo/interp/interp.go:236 +0x4bb
main.main()
	/Users/marc/go/src/github.com/containous/dyngo/dyngo.go:50 +0x334
```

--- a/_test/comp0.go
+++ b/_test/comp0.go
@@ -1,0 +1,8 @@
+package main
+
+func main() {
+	println(2 < 2.4)
+}
+
+// Output:
+// true

--- a/cmd/genop/genop.go
+++ b/cmd/genop/genop.go
@@ -156,6 +156,87 @@ func {{$name}}(n *Node) {
 	}
 }
 {{end}}
+{{range $name, $op := .Comparison}}
+func {{$name}}(n *Node) {
+	tnext := getExec(n.tnext)
+	typ := n.typ.TypeOf()
+
+	switch {
+	case typ.Kind() == reflect.String:
+		v0 := genValueString(n.child[0])
+		v1 := genValueString(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) {{$op.Name}} v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) {{$op.Name}} v1(f))
+				return tnext
+			}
+		}
+	case isFloat(n.child[0].typ) || isFloat(n.child[1].typ):
+		v0 := genValueFloat(n.child[0])
+		v1 := genValueFloat(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) {{$op.Name}} v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) {{$op.Name}} v1(f))
+				return tnext
+			}
+		}
+	case isUint(n.child[0].typ) || isUint(n.child[1].typ):
+		v0 := genValueUint(n.child[0])
+		v1 := genValueUint(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) {{$op.Name}} v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) {{$op.Name}} v1(f))
+				return tnext
+			}
+		}
+	default:
+		v0 := genValueInt(n.child[0])
+		v1 := genValueInt(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) {{$op.Name}} v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) {{$op.Name}} v1(f))
+				return tnext
+			}
+		}
+	}
+}
+{{end}}
 `
 
 // Op define operator name and properties
@@ -191,6 +272,14 @@ func main() {
 		"IncDec": map[string]Op{
 			"inc": {Name: "+"},
 			"dec": {Name: "-"},
+		},
+		"Comparison": map[string]Op{
+			"equal":        {Name: "=="},
+			"greater":      {Name: ">"},
+			"greaterEqual": {Name: ">="},
+			"lower":        {Name: "<"},
+			"lowerEqual":   {Name: "<="},
+			"notEqual":     {Name: "!="},
 		},
 	}
 	if err = parse.Execute(b, data); err != nil {

--- a/interp/ast.go
+++ b/interp/ast.go
@@ -200,12 +200,14 @@ const (
 	Defer
 	Equal
 	Greater
+	GreaterEqual
 	GetFunc
 	GetIndex
 	Inc
 	Land
 	Lor
 	Lower
+	LowerEqual
 	Mul
 	MulAssign
 	Negate
@@ -489,6 +491,8 @@ func (interp *Interpreter) ast(src, name string) (string, *Node, error) {
 				action = AndNot
 			case token.EQL:
 				action = Equal
+			case token.GEQ:
+				action = GreaterEqual
 			case token.GTR:
 				action = Greater
 			case token.LAND:
@@ -497,6 +501,8 @@ func (interp *Interpreter) ast(src, name string) (string, *Node, error) {
 			case token.LOR:
 				kind = LorExpr
 				action = Lor
+			case token.LEQ:
+				action = LowerEqual
 			case token.LSS:
 				action = Lower
 			case token.MUL:

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -387,7 +387,7 @@ func (interp *Interpreter) Cfg(root *Node) ([]*Node, error) {
 				if n.child[0].sym == nilSym || n.child[1].sym == nilSym {
 					n.gen = isNil
 				}
-			case Greater, Lower:
+			case Greater, GreaterEqual, Lower, LowerEqual:
 				n.typ = scope.getType("bool")
 			default:
 				n.typ, err = nodeType(interp, scope, n)

--- a/interp/interp_test.go
+++ b/interp/interp_test.go
@@ -1413,6 +1413,25 @@ func main() {
 	// 9
 }
 
+func Example_comp0() {
+	src := `
+package main
+
+func main() {
+	println(2 < 2.4)
+}
+`
+	i := interp.New(interp.Opt{Entry: "main"})
+	i.Use(stdlib.Value, stdlib.Type)
+	_, err := i.Eval(src)
+	if err != nil {
+		panic(err)
+	}
+
+	// Output:
+	// true
+}
+
 func Example_const0() {
 	src := `
 package main

--- a/interp/op.go
+++ b/interp/op.go
@@ -705,3 +705,483 @@ func inc(n *Node) {
 		}
 	}
 }
+
+func equal(n *Node) {
+	tnext := getExec(n.tnext)
+	typ := n.typ.TypeOf()
+
+	switch {
+	case typ.Kind() == reflect.String:
+		v0 := genValueString(n.child[0])
+		v1 := genValueString(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) == v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) == v1(f))
+				return tnext
+			}
+		}
+	case isFloat(n.child[0].typ) || isFloat(n.child[1].typ):
+		v0 := genValueFloat(n.child[0])
+		v1 := genValueFloat(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) == v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) == v1(f))
+				return tnext
+			}
+		}
+	case isUint(n.child[0].typ) || isUint(n.child[1].typ):
+		v0 := genValueUint(n.child[0])
+		v1 := genValueUint(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) == v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) == v1(f))
+				return tnext
+			}
+		}
+	default:
+		v0 := genValueInt(n.child[0])
+		v1 := genValueInt(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) == v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) == v1(f))
+				return tnext
+			}
+		}
+	}
+}
+
+func greater(n *Node) {
+	tnext := getExec(n.tnext)
+	typ := n.typ.TypeOf()
+
+	switch {
+	case typ.Kind() == reflect.String:
+		v0 := genValueString(n.child[0])
+		v1 := genValueString(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) > v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) > v1(f))
+				return tnext
+			}
+		}
+	case isFloat(n.child[0].typ) || isFloat(n.child[1].typ):
+		v0 := genValueFloat(n.child[0])
+		v1 := genValueFloat(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) > v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) > v1(f))
+				return tnext
+			}
+		}
+	case isUint(n.child[0].typ) || isUint(n.child[1].typ):
+		v0 := genValueUint(n.child[0])
+		v1 := genValueUint(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) > v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) > v1(f))
+				return tnext
+			}
+		}
+	default:
+		v0 := genValueInt(n.child[0])
+		v1 := genValueInt(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) > v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) > v1(f))
+				return tnext
+			}
+		}
+	}
+}
+
+func greaterEqual(n *Node) {
+	tnext := getExec(n.tnext)
+	typ := n.typ.TypeOf()
+
+	switch {
+	case typ.Kind() == reflect.String:
+		v0 := genValueString(n.child[0])
+		v1 := genValueString(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) >= v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) >= v1(f))
+				return tnext
+			}
+		}
+	case isFloat(n.child[0].typ) || isFloat(n.child[1].typ):
+		v0 := genValueFloat(n.child[0])
+		v1 := genValueFloat(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) >= v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) >= v1(f))
+				return tnext
+			}
+		}
+	case isUint(n.child[0].typ) || isUint(n.child[1].typ):
+		v0 := genValueUint(n.child[0])
+		v1 := genValueUint(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) >= v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) >= v1(f))
+				return tnext
+			}
+		}
+	default:
+		v0 := genValueInt(n.child[0])
+		v1 := genValueInt(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) >= v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) >= v1(f))
+				return tnext
+			}
+		}
+	}
+}
+
+func lower(n *Node) {
+	tnext := getExec(n.tnext)
+	typ := n.typ.TypeOf()
+
+	switch {
+	case typ.Kind() == reflect.String:
+		v0 := genValueString(n.child[0])
+		v1 := genValueString(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) < v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) < v1(f))
+				return tnext
+			}
+		}
+	case isFloat(n.child[0].typ) || isFloat(n.child[1].typ):
+		v0 := genValueFloat(n.child[0])
+		v1 := genValueFloat(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) < v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) < v1(f))
+				return tnext
+			}
+		}
+	case isUint(n.child[0].typ) || isUint(n.child[1].typ):
+		v0 := genValueUint(n.child[0])
+		v1 := genValueUint(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) < v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) < v1(f))
+				return tnext
+			}
+		}
+	default:
+		v0 := genValueInt(n.child[0])
+		v1 := genValueInt(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) < v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) < v1(f))
+				return tnext
+			}
+		}
+	}
+}
+
+func lowerEqual(n *Node) {
+	tnext := getExec(n.tnext)
+	typ := n.typ.TypeOf()
+
+	switch {
+	case typ.Kind() == reflect.String:
+		v0 := genValueString(n.child[0])
+		v1 := genValueString(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) <= v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) <= v1(f))
+				return tnext
+			}
+		}
+	case isFloat(n.child[0].typ) || isFloat(n.child[1].typ):
+		v0 := genValueFloat(n.child[0])
+		v1 := genValueFloat(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) <= v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) <= v1(f))
+				return tnext
+			}
+		}
+	case isUint(n.child[0].typ) || isUint(n.child[1].typ):
+		v0 := genValueUint(n.child[0])
+		v1 := genValueUint(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) <= v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) <= v1(f))
+				return tnext
+			}
+		}
+	default:
+		v0 := genValueInt(n.child[0])
+		v1 := genValueInt(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) <= v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) <= v1(f))
+				return tnext
+			}
+		}
+	}
+}
+
+func notEqual(n *Node) {
+	tnext := getExec(n.tnext)
+	typ := n.typ.TypeOf()
+
+	switch {
+	case typ.Kind() == reflect.String:
+		v0 := genValueString(n.child[0])
+		v1 := genValueString(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) != v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) != v1(f))
+				return tnext
+			}
+		}
+	case isFloat(n.child[0].typ) || isFloat(n.child[1].typ):
+		v0 := genValueFloat(n.child[0])
+		v1 := genValueFloat(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) != v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) != v1(f))
+				return tnext
+			}
+		}
+	case isUint(n.child[0].typ) || isUint(n.child[1].typ):
+		v0 := genValueUint(n.child[0])
+		v1 := genValueUint(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) != v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) != v1(f))
+				return tnext
+			}
+		}
+	default:
+		v0 := genValueInt(n.child[0])
+		v1 := genValueInt(n.child[1])
+		if n.fnext != nil {
+			fnext := getExec(n.fnext)
+			n.exec = func(f *Frame) Builtin {
+				if v0(f) != v1(f) {
+					return tnext
+				}
+				return fnext
+			}
+		} else {
+			i := n.findex
+			n.exec = func(f *Frame) Builtin {
+				f.data[i].SetBool(v0(f) != v1(f))
+				return tnext
+			}
+		}
+	}
+}

--- a/interp/run.go
+++ b/interp/run.go
@@ -32,10 +32,12 @@ var builtin = [...]BuiltinGenerator{
 	Equal:        equal,
 	GetFunc:      getFunc,
 	Greater:      greater,
+	GreaterEqual: greaterEqual,
 	Inc:          inc,
 	Land:         land,
 	Lor:          lor,
 	Lower:        lower,
+	LowerEqual:   lowerEqual,
 	Mul:          mul,
 	MulAssign:    mulAssign,
 	Negate:       negate,
@@ -773,75 +775,6 @@ func negate(n *Node) {
 	}
 }
 
-func equal(n *Node) {
-	i := n.findex
-	value0 := genValue(n.child[0])
-	value1 := genValue(n.child[1])
-	tnext := getExec(n.tnext)
-
-	if n.fnext == nil {
-		n.exec = func(f *Frame) Builtin {
-			f.data[i].SetBool(value0(f).Interface() == value1(f).Interface())
-			return tnext
-		}
-	} else {
-		fnext := getExec(n.fnext)
-
-		n.exec = func(f *Frame) Builtin {
-			if value0(f).Interface() == value1(f).Interface() {
-				return tnext
-			}
-			return fnext
-		}
-	}
-}
-
-func notEqual(n *Node) {
-	i := n.findex
-	value0 := genValue(n.child[0])
-	value1 := genValue(n.child[1])
-	tnext := getExec(n.tnext)
-
-	if n.fnext == nil {
-		n.exec = func(f *Frame) Builtin {
-			f.data[i].SetBool(value0(f).Interface() != value1(f).Interface())
-			return tnext
-		}
-	} else {
-		fnext := getExec(n.fnext)
-
-		n.exec = func(f *Frame) Builtin {
-			if value0(f).Interface() != value1(f).Interface() {
-				return tnext
-			}
-			return fnext
-		}
-	}
-}
-
-func greater(n *Node) {
-	i := n.findex
-	value0 := genValue(n.child[0])
-	value1 := genValue(n.child[1])
-	tnext := getExec(n.tnext)
-
-	if n.fnext == nil {
-		n.exec = func(f *Frame) Builtin {
-			f.data[i].SetBool(value0(f).Int() > value1(f).Int())
-			return tnext
-		}
-	} else {
-		fnext := getExec(n.fnext)
-
-		n.exec = func(f *Frame) Builtin {
-			if value0(f).Int() > value1(f).Int() {
-				return tnext
-			}
-			return fnext
-		}
-	}
-}
-
 func land(n *Node) {
 	value0 := genValue(n.child[0])
 	value1 := genValue(n.child[1])
@@ -881,28 +814,6 @@ func lor(n *Node) {
 		i := n.findex
 		n.exec = func(f *Frame) Builtin {
 			f.data[i].SetBool(value0(f).Bool() || value1(f).Bool())
-			return tnext
-		}
-	}
-}
-
-func lower(n *Node) {
-	value0 := genValue(n.child[0])
-	value1 := genValue(n.child[1])
-	tnext := getExec(n.tnext)
-
-	if n.fnext != nil {
-		fnext := getExec(n.fnext)
-		n.exec = func(f *Frame) Builtin {
-			if value0(f).Int() < value1(f).Int() {
-				return tnext
-			}
-			return fnext
-		}
-	} else {
-		i := n.findex
-		n.exec = func(f *Frame) Builtin {
-			f.data[i].SetBool(value0(f).Int() < value1(f).Int())
 			return tnext
 		}
 	}

--- a/interp/type.go
+++ b/interp/type.go
@@ -518,6 +518,18 @@ func isInt(t *Type) bool {
 	return false
 }
 
+func isUint(t *Type) bool {
+	typ := t.TypeOf()
+	if typ == nil {
+		return false
+	}
+	switch typ.Kind() {
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return true
+	}
+	return false
+}
+
 func isFloat(t *Type) bool {
 	typ := t.TypeOf()
 	if typ == nil {

--- a/interp/value.go
+++ b/interp/value.go
@@ -130,3 +130,8 @@ func genValueFloat(n *Node) func(*Frame) float64 {
 	}
 	return nil
 }
+
+func genValueString(n *Node) func(*Frame) string {
+	value := genValue(n)
+	return func(f *Frame) string { return value(f).String() }
+}

